tests/system/python/api/test_statistics.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test Statistics & Statistics history REST API """
+
+
+import http.client
+import json
+import time
+from collections import Counter
+import pytest
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+KEYS = {'DISCARDED', 'PURGED', 'history_ts', 'READINGS', 'UNSENT', 'UNSNPURGED', 'BUFFERED'}
+
+
+class TestStatistics:
+
+    def test_default_statistics(self, reset_and_start_foglamp, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/statistics')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert 6 == len(jdoc)
+        keys = [key['key'] for key in jdoc]
+        assert Counter(['BUFFERED', 'DISCARDED', 'PURGED', 'READINGS', 'UNSENT',
+                        'UNSNPURGED']) == Counter(keys)
+
+    def test_default_statistics_history(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/statistics/history')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert 15 == jdoc['interval']
+        assert {} == jdoc['statistics'][0]
+
+    def test_statistics_history_with_stats_collector_schedule(self, foglamp_url, wait_time):
+        # wait for sometime for stats collector schedule to start
+        time.sleep(wait_time * 3)
```

Comment:
let's not keep waiting for 15 seconds, but run stats collector every 2 -5 seconds for test

Reply:
And  with south plugin is enabled before, check stats and history are collected for that. (as the last test)

Reply:
Its not the responsibility of System REST API tests; as these scenarios are covered in `e2e` tests

Reply:
for stats yes, but not the stats history.

Reply:
Added

Reply:
def test_statistics_history_with_stats_collector_schedule(self, foglamp_url, wait_time, retries):
          time.sleep(wait_time * retries)